pkg/scheduler/plugins/predicates/predicates.go

```diff
@@ -547,8 +548,15 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 		}
 
 		for _, val := range api.RegisteredDevices {
-			if devices, ok := node.Others[val].(api.Devices); ok {
-				code, msg, err := devices.FilterNode(task.Pod)
+			if dev, ok := node.Others[val].(api.Devices); ok {
+				if dev == nil {
```

Comment:
Under what condition, the `dev` is nil?

Reply:
It happens when you enable the device in predicates.VGPUEnable arguments, but filter out some GPU nodes using "nodeSelector" when you apply vgpu-device-plugin. Then node.Others["vgpu"] will be nil for these filtered-out nodes.